### PR TITLE
Scala versioning

### PR DESCRIPTION
**Pull request by pauldraper**
Objectives: Simplify, preserving ability to have multiple Scala versions, make a single Scala version very easy.

* Remove cross-Scala versioning from rules
* Add typelevel Scala test
* Consolidate bootstrapping scala compilation into basic_scala_library
* Create @scala (default), @scala_annex_scala_2_12, and //external:scala_annex_scala

### CODE REVIEW

File: rules/build.bzl [hunk 1]
```diff
@@ -93,7 +106,8 @@ annex_scala_library = rule(
         "srcs": attr.label_list(allow_files = [".scala", ".java"]),
         "deps": attr.label_list(),
         "exports": attr.label_list(),
-        "scala": attr.label_list(
+        "scala": attr.label(
```
Comment (andyscott) [inline rules/build.bzl]: 👍

**Comment by andyscott** [comment]
Looks good so far-- I need a bit more time to read through the whole thing. Additionally this needs a rebase with master.

**Comment by pauldraper** [comment]
Any more thoughts before I merge?

**Comment by andyscott** [comment]
LGTM

**Pull request merged**